src/app/components/AutoComplete.svelte

```diff
@@ -0,0 +1,176 @@
+<script lang="ts">
+  // TODO: the following interface should be removed once #33 is done
+  interface dataObj {
+    bcp47Tag: string;
+    language: string;
+  }
+  enum keyboardKey {
+    Up = "ArrowUp",
+    Down = "ArrowDown",
+    Enter = "Enter",
+  }
+  export let label = "";
+  export let subtext = "";
+  // Testing Data
+  export let results: dataObj[] = [];
+  // To store filtered array
+  let filtered: dataObj[] = [];
+  // To store selected language
+  let selected: string = "";
+  // Toggle to show search list
+  let show = false;
+  // Index of Previous focus element
+  let prevIndex = -1;
+  // Index of focus element
+  let index = -1;
+
+  // Input to search
+  function onChange(event: Event) {
+    show = true;
+    filtered = results.filter((item: any) => {
+      const target = event.target as HTMLTextAreaElement;
+      const a = new RegExp("^" + target.value.toUpperCase());
+      return a.test(item.language.toUpperCase());
+    });
+  }
+
+  function closeSuggestion() {
+    show = false;
+  }
+
+  // If anywhere on the screen is clicked
+  // Derived from https://github.com/rster2002/svelte-outside-click
+  function clickOutside(node: HTMLUListElement, onEventFunction: Function) {
+    const handleClick = (event: MouseEvent) => {
+      const path = event.composedPath();
+      if (!path.includes(node)) {
+        onEventFunction();
+      }
+    };
+    document.addEventListener("click", handleClick);
+    return {
+      destroy() {
+        document.removeEventListener("click", handleClick);
+      },
+    };
+  }
+
+  // On select item in list
+  function selectedList(data: dataObj) {
+    show = false;
+    selected = data.language;
+    subtext = data.bcp47Tag;
+  }
+
+  const mod = (a: number, n: number) => {
+    return a - n * Math.floor(a / n);
+  };
+
+  // Up/Down arraow
+  function handleKeydown({ key }: KeyboardEvent) {
+    if (key === keyboardKey.Down) {
+      prevIndex = index;
+      index += 1;
+    } else if (key === keyboardKey.Up) {
+      if (index == -1) {
+        index = -1;
+      } else {
+        prevIndex = index;
+        index -= 1;
+      }
+    } else if (key === keyboardKey.Enter) {
+      selectedList(filtered[index]);
+    } else {
+      /* Not a key we care about */
+      return;
+    }
+    const allItems = [
+      ...document.getElementsByClassName("autocomplete__suggestion-item"),
+    ];
```

Comment:
I just noticed that this clever trick isn't actually needed. `allItems` is created to get the `.length` to bound the modulus. However, [`document.getElementsByClassName()`](https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementsByClassName) returns an `HTMLCollection`, which [has a `.length` property](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection#Properties)! The `allItems` array is not used otherwise, so just get its length!

 ```suggestion
    const numberOfSuggestions = document.getElementsByClassName("autocomplete__suggestion-item").length;
```

Reply:
I like this suggestion and it also omit the use of spread operator